src/main/java/com/cburch/logisim/gui/start/Startup.java

```diff
@@ -145,388 +155,534 @@ static void doPrint(File file) {
+      return RC.QUIT;
+    }
+    final var file = opt.getValue();
+    startup.templFile = new File(file);
+    String errMsg = null;
+    if (!startup.templFile.exists()) errMsg = S.get("templateMissingError", file);
+    if (errMsg == null && !startup.templFile.canRead()) errMsg = S.get("templateCannotReadError", file);
+    if (errMsg != null) return RC.WARN;  // FIXME: shouldn't we quit in such case?
+    return RC.OK;
+  }
+
+  private static RC handleArgNoSplash(Startup startup, Option opt) {
+    startup.showSplash = false;
+    return RC.OK;
+  }
+
+  private static RC handleArgTestVector(Startup startup, Option opt) {
+    startup.circuitToTest = opt.getValues()[0];
+    startup.testVector = opt.getValues()[1];
+    startup.showSplash = false;
+    startup.exitAfterStartup = true;
+    // This is to test a test bench. It will return 0 or 1 depending on if the tests pass or not.
+    return RC.OK;
+  }
+
+  private static RC handleArgTestFpgaImpl(Startup startup, Option opt) {
+    final var optArgs = opt.getValues();
+
+    // already handled above
+    startup.testCircuitImpPath = optArgs[0];
+    startup.testCircuitImpMapFile = optArgs[1];
+    startup.testCircuitImpName = optArgs[2];
+    startup.testCircuitImpBoard = optArgs[3];
+
+    if (optArgs.length > 4) {
+      try {
+        startup.testTickFrequency = Integer.parseUnsignedInt(optArgs[4]);
+      } catch (NumberFormatException ignored) {
+        // FIXME: do nothing, but that's not the best error handlong
+      }
+      if (optArgs.length > 5) {
+        startup.testCircuitHdlOnly = optArgs[5].equalsIgnoreCase("HDLONLY");
+      }
+    }
+
+    startup.doFpgaDownload = true;
+    startup.showSplash = false;
+    startup.filesToOpen.add(new File(startup.testCircuitImpPath));
+    return RC.OK;
+  }
+
+  private static RC handleArgTestCircuit(Startup startup, Option opt) {
+    final var fileName = opt.getValue();
+    startup.testCircuitPathInput = fileName;
+    startup.filesToOpen.add(new File(fileName));
+    startup.showSplash = false;
+    startup.exitAfterStartup = true;
+    return RC.OK;
+  }
+
+  private static RC handleArgTestCircGen(Startup startup, Option opt) {
+    final var optArgs = opt.getValues();
+    // This is to test the XML consistency over different version of the Logisim
+    // This is the input path of the file to open
+    startup.testCircPathInput = optArgs[0];
+    startup.filesToOpen.add(new File(startup.testCircPathInput));
+    // This is the output file's path. The comparaison shall be done between the  testCircPathInput and the testCircPathOutput
+    startup.testCircPathOutput = optArgs[1];
+    startup.showSplash = false;
+    startup.exitAfterStartup = true;
+    return RC.OK;
+  }
+
+  private static RC handleArgCircuit(Startup startup, Option opt) {
+    startup.circuitToTest = opt.getValue();
+    return RC.OK;
+  }
+
+  private static RC handleArgQuesta(Startup startup, Option opt) {
+    try {
```

Comment:
This command line option I would also remove, as:

1) It is not required to use Questasim, also Modelsim (the cheap version of Questasim) will do the trick.
2) When switched on, and no path given logisim will all the time complain and is unworkable.

Reply:
Removed.